portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,46 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/**
- * Adds a random greeting to the page.
- */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+/* Shows about text when the "About" button is pressed. */
+function showAbout () {
+    const aboutText = "I'm a rising junior at UCLA, majoring in Linguistics and Computer Science (that's one major)! In my free time, I enjoy reading and playing the piano. \n\n Recently, I've started learning to cook (I'm going to need this for my college apartment next year) and playing Stardew Valley (turns out video games are a great way to stay connected to friends during this time)!";
+    const aboutContainer = document.getElementById('aboutSpace');
+    //Change the text in the aboutSpace element 
+    if (aboutContainer.innerText === "") {
+        //Change inner text to about text, and div class to CSS about class
+        aboutContainer.innerText = aboutText;
+        aboutContainer.className = "background";
+    }
+    else {
+        aboutContainer.innerText = "";
+        aboutContainer.className= "";
+    }
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/* Shows pictures when the "Pictures" button is pressed. */
+function showPics() {
+    const imgs = ["ucla1.JPG", "ucla2.jpg.JPG", "book.JPG", "harrypotter.JPG"]
+    const srcBase = "https://8080-0ba309ed-2f3c-4b55-83b6-7741ff94bec3.us-west1.cloudshell.dev/base"
+    const picPath = "https://8080-0ba309ed-2f3c-4b55-83b6-7741ff94bec3.us-west1.cloudshell.dev/images/"
+    
+    //Random int from 0 - size of images array
+    const randInt = (Math.floor(Math.random() * imgs.length));
+    const randImg = imgs[randInt];
+    
+    const div = document.getElementById('picSpace');
+    const picContainer = div.getElementsByTagName('img')[0];
```

Comment:
with stricter null-checks, you would need to first ensure that getElementsByTagName returns an array with > 0 elements before accessing the first one, so it might be good practice to do so here.